General updates to language changing in JASP

JorisGoosen opened the pull request:
- Also fix some whitespace
- Fixes https://github.com/jasp-stats/jasp-issues/issues/1088 (analysis results headers dont change on language change)
- Javascript should use just the proper title from Analysis directly instead of the one From R. So jaspResults does not need to send it either.
- pause and resume engines for languagechange to avoid running stuff while it is still going on
- Also added a queue for javascript commands to the results, this because the commands are otherwise silently ignored during the small window of time where the webeengine is reloading during languagechange.
- And removed some legacy JS code, pretty sure this doesn't affect anything but would appreciate if Tim glances at it suavely.

JorisGoosen committed:
General updates to language changing in JASP
- Also fix some whitespace
- Fixes https://github.com/jasp-stats/jasp-issues/issues/1088 (analysis results headers dont change on language change)
- Javascript should use just the proper title from Analysis directly instead of the one From R. So jaspResults does not need to send it either.
- pause and resume engines for languagechange to avoid running stuff while it is still going on
- Also added a queue for javascript commands to the results, this because the commands are otherwise silently ignored during the small window of time where the webeengine is reloading during languagechange.
- And removed some legacy JS code, pretty sure this doesn't affect anything but would appreciate if Tim glances at it suavely.

JorisGoosen commented:
Pfoewey, the problems here were quite a bit worse and hard to untangle than I expected when I started.
But I think Ive got it all straightened out now.

To make life easier for you poor reviewers I've turned on some nightlies that are building as I type.

Tim, I added you mostly for the javascript to check that it was ok for me to remove that stuff. Im pretty sure it could do with a bigger cleanup though. Some other day.

Frans, I can take you through the changes and why I made them tomorrow if you like?

Hi Bruno!

JorisGoosen commented:
I think maybe the progressbar in the results  might need some JS love though.

TimKDJ commented on Desktop/html/js/analysis.js:
```diff
@@ -632,15 +593,8 @@ JASPWidgets.AnalysisView = JASPWidgets.View.extend({
 
-		var results = this.model.get("results");
-
-		// once everything becomes jaspResults this is always an object and the following can be removed
-		var progress = this.model.get("progress")
-		if (typeof progress == "number")
-			this.model.set("progress", { value: progress, label: "" })
-		else if (!progress)
-			this.model.set("progress", { value: -1, label: "" })
-		// up to here
```
I recall from a previous attempt to remove this, that `progress` is not always an object even now with jaspResults. So I'd keep this in here

JorisGoosen replied:
Or I can push a fix to jaspResults to always make it an object

JorisGoosen replied:
I think, if it isn't an object it will be `null` that should be fairly easy to check for.

TimKDJ replied:
We at least have to be mindful of it being null in:
https://github.com/jasp-stats/jasp-desktop/blob/0db025f5aa07b9a192599c60c407f1ec23fd7b74/Desktop/analysis/analysis.h#L80
and in the engine.cpp there seem to be a few `_progress					= -1;` hardcoded, although it could well be that the analysis render is never called in that case.

JorisGoosen replied:
Well, I didn't see those, but maybe I didnt look hard enough. I can just change those "-1" to null and then check for it in js.

JorisGoosen replied:
So I think the new commit should fix the progressbar blemishes.

JorisGoosen committed:
Fixes progressbar stuff

TimKDJ commented on Desktop/html/js/analysis.js:
```diff
@@ -596,9 +596,10 @@ JASPWidgets.AnalysisView = JASPWidgets.View.extend({
 
-		if (results == "" || results == null) {
+		if (results === "" || results === null) {
 			progress = this.model.get("progress");
-			if (progress.value > -1)
+
+			if (progress !== null && progress.value > -1)
 				this.updateProgressbarInResults();
 			return this;
 		}
```
If you want you can delete these lines, as they won't be used anymore. They belong to the old style analysis only sending progress updates without the results being included. I don't think this is possible anymore with jaspResults

JorisGoosen replied:
Yeah but https://github.com/jasp-stats/INTERNAL-jasp/issues/1159 could make use of that really well.
It should also avoid some of the slowness thats been observed in the webengine during progressbar phase (because it keeps rerendering the results despite them not changing).

TimKDJ commented:
So I've done some testing and the language changing seems to work ok, so that's great. I did notice that on a few occasions the ribbon got stuck when messing around in analyses -- sometimes in combination with an analysis itself getting stuck. Not sure if this problem is introduced by the PR though. Will try to get a reproduction path, but messing around with the Bayesian ANOVA and "Model averaged posteriors" seemed to do the trick a few times.
![ribbons-stuck](https://user-images.githubusercontent.com/18737241/105489056-e8869400-5cb2-11eb-815a-00b626fdc7f9.png)

JorisGoosen commented:
I'll have a look at the ribbon-loading problem.

JorisGoosen committed:
- Make sure Bayesian ANOVA doesnt break all modules when running and changing language
- Load all modules at the same time for faster startup and less loading symbols. Might fix https://github.com/jasp-stats/INTERNAL-jasp/issues/1224 as well

JorisGoosen commented:
I think now it won't break anymore and it should all work a bit better.

SamanthaS1 commented:
Hi there, I was just wondering how I would be able to find the missing values in the descriptive statistics chart in jasp. So in other words, the problem I've come across is that it's not giving me the options for descriptive statistics in order to analyze values such as median, mode, variance and range. So my question is, how do I add the missing values without the option of adding them? Am I supposed to request this option or some kind of update? The main issue is that it's just not letting me scroll down to where I can check the values that I need for the analysis. If someone could get back to me on this I'd appreciate it,Thanks.

JorisGoosen commented:
Hello @SamanthaS1 this is not quite the right place to ask these kind of questions, I suggest you open an issue via https://github.com/jasp-stats/jasp-issues/issues/new/choose or ask on the [cogsci forum](https://forum.cogsci.nl/categories/jasp-bayesfactor)
